src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,180 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(tail, value, null);
+        if (tail == null) {
+            head = newNode;
+        } else {
+            tail.next = newNode;
+        }
+        tail = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkAddIndex(index);
+        Node<T> newNode;
+        if (tail == null) {
+            newNode = new Node<>(null, value, null);
+            head = tail = newNode;
+        } else if (index == 0) {
+            newNode = new Node<>(null, value, head);
+            head.prev = newNode;
+            head = newNode;
+        } else if (index == size) {
+            newNode = new Node<>(tail, value, null);
+            tail.next = newNode;
+            tail = newNode;
+        } else {
+            Node<T> current = findNode(index);
+            newNode = new Node<>(current.prev, value, current);
+            current.prev.next = newNode;
+            current.prev = newNode;
+
+        }
+        size++;
+        if (tail == newNode) {
+            head.prev = newNode;
```

Comment:
This line is unnecessary as the tail is already being updated in other parts of the code properly.